plugins-compat-tester/src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -37,48 +39,75 @@
 import javax.xml.xpath.XPathExpression;
 import javax.xml.xpath.XPathExpressionException;
 import javax.xml.xpath.XPathFactory;
+import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URL;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
 import javax.xml.xpath.XPath;
 
 /**
- * Utility class providing business for retrieving plugin scm data
+ * Utility class providing business for retrieving plugin POM data
  * @author Frederic Camblor
  */
 public class PluginRemoting {
 
-	private String hpiRemoteUrl;
-	
-	public PluginRemoting(String hpiRemoteUrl){
-		this.hpiRemoteUrl = hpiRemoteUrl;
-	}
+    private String hpiRemoteUrl;
+    private File pomFile;
+
+    public PluginRemoting(String hpiRemoteUrl){
+        this.hpiRemoteUrl = hpiRemoteUrl;
+    }
+
+    public PluginRemoting(File pomFile){
+        this.pomFile = pomFile;
+    }
 	
-	private String retrievePomContent() throws PluginSourcesUnavailableException{
-		try {
-			URL pluginUrl = new URL(hpiRemoteUrl);
-			ZipInputStream zin = new ZipInputStream(pluginUrl.openStream());
```

Comment:
[diff for reviewers](https://github.com/jenkinsci/plugin-compat-tester/pull/30/files?w=1#diff-b0ba84ccff4945a14be98195821b8b72R75)